app/lib/features/space/sheets/link_chat_sheet.dart

```diff
@@ -0,0 +1,224 @@
+import 'package:acter/common/widgets/spaces/select_space_form_field.dart';
+import 'package:acter/features/chat/providers/chat_providers.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class LinkChatPage extends ConsumerStatefulWidget {
+  final String? parentSpaceId;
+
+  const LinkChatPage({super.key, required this.parentSpaceId});
+
+  @override
+  ConsumerState<LinkChatPage> createState() => _LinkChatPageConsumerState();
+}
+
+class _LinkChatPageConsumerState extends ConsumerState<LinkChatPage> {
+  final TextEditingController searchTextEditingController =
+      TextEditingController();
+  final List<String> subChatsIds = [];
+
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      selectParentSpaceData();
+    });
+    super.initState();
+  }
+
+  //Select parent space data
+  void selectParentSpaceData() {
+    if (widget.parentSpaceId != null) {
+      final notifier = ref.read(selectedSpaceIdProvider.notifier);
+      notifier.state = widget.parentSpaceId;
+    }
+  }
+
+  //Fetch known sub-chats list of selected parent space
+  void fetchKnownSubChatsData() {
+    final selectedParentSpaceId = ref.watch(selectedSpaceIdProvider);
+    if (selectedParentSpaceId == null) return;
+    final sp = ref.watch(spaceRelationsOverviewProvider(selectedParentSpaceId));
+    sp.when(
+      data: (space) {
+        subChatsIds.clear();
+        for (int i = 0; i < space.knownChats.length; i++) {
+          subChatsIds.add(space.knownChats[i].getRoomId().toString());
+        }
+      },
+      error: (e, s) => Container(),
+      loading: () => Container(),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+
+    //Fetch known sub-chats list of selected parent space
+    fetchKnownSubChatsData();
+
+    return SideSheet(
+      header: 'Link Sub-Chat',
+      body: SizedBox(
+        height: size.height - 120,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            parentSpaceSelector(context, ref),
+            Search(
+              onChanged: (value) => ref
+                  .read(chatSearchValueProvider.notifier)
+                  .update((state) => value),
+              searchController: searchTextEditingController,
+            ),
+            Expanded(child: chatsList(context, ref)),
+          ],
+        ),
+      ),
+    );
+  }
+
+  //Parent space selector
+  Widget parentSpaceSelector(BuildContext context, WidgetRef ref) {
+    return const Padding(
+      padding: EdgeInsets.all(16),
+      child: SelectSpaceFormField(
+        canCheck: 'CanLinkSpaces',
+        mandatory: true,
+        title: 'Parent space',
+        emptyText: 'optional parent space',
```

Comment:
It's not optional now, is it ;) .

```suggestion
        emptyText: '(none selected)`,
```